src/EmployeesAdministrationApp.java

```diff
@@ -151,21 +157,16 @@ private static void addEmployee(Object employee, String message) {
 				}
 			}
 		}
-		Map<String, Map<String, Object>> employeeDetailsJson = new HashMap<>();
-		employeeDetailsJson.put("employee", employeeDetails);
-		JSONObject employeeDetailsJsonObject = new JSONObject(employeeDetailsJson);
-		listOfEmployees.add(employeeDetailsJsonObject);
+		listOfEmployees.add((HashMap<String, Object>) employeeDetails);
 		System.out.println("The employee was added to your list.");
 	}
 
 	private static int getValidEmployeeId(Object employee) {
 		System.out.println("Please type in the employeeId of the new employee: ");
 		Scanner scan = new Scanner(System.in);
 		int idOfTheEmployee = scan.nextInt();
-		for (Object emp : listOfEmployees) {
-			HashMap<String, Object> object = (HashMap<String, Object>) ((HashMap<String, Object>) emp).get("employee");
-			JSONObject obj = new JSONObject(object);
-			if (Integer.parseInt(String.valueOf(obj.get("EmployeeId"))) == idOfTheEmployee) {
+		for (HashMap<String, Object> emp : listOfEmployees) {
```

Comment:
here you should iterate through hash map, not through a list of hashmaps

Reply:
I changed the list of employees from List to HashMap